lib/finite_machine/observer.rb

```diff
@@ -40,15 +43,19 @@ def self.cleanup_callback_queue
     #
     # @api public
     def initialize(machine)
-      @machine = machine
-      @hooks   = Hooks.new
+      @machine   = machine
+      @hooks     = Hooks.new
+      @object_id = nil
 
       @machine.subscribe(self)
-      ObjectSpace.define_finalizer(self, self.class.cleanup_callback_queue)
     end
 
     def callback_queue
-      @callback_queue ||= MessageQueue.new
+      return @callback_queue if instance_variable_defined?(:@callback_queue)
+
+      @callback_queue = MessageQueue.new
+      @object_id = SecureRandom.uuid
+      ObjectSpace.define_finalizer(@object_id, cleanup_callback_queue)
```

Comment:
@vkononov This method needs to return `@callback_queue`

```
return @callback_queue if instance_variable_defined?(:@callback_queue)

      @callback_queue = MessageQueue.new
      @object_id = SecureRandom.uuid
      ObjectSpace.define_finalizer(@object_id, cleanup_callback_queue)
      @callback_queue
```

Add `@callback_queue` to return from this method fixes pipeline